### PR TITLE
add MaxDataDisksPerVM to prevent trying to mount too many disks on a single node

### PR DESCRIPTION
**Pull request by jim-minter**
cf #945

### CODE REVIEW

File: pkg/config/derive.go [hunk 1]
```diff
@@ -139,3 +139,87 @@ func (derived) StatsdArgs(cs *api.OpenShiftManagedCluster) ([]interface{}, error
...
+	for i := range cs.Properties.AgentPoolProfiles {
+		if cs.Properties.AgentPoolProfiles[i].Role != api.AgentPoolProfileRoleCompute {
+			continue
+		}
+
+		if app != nil {
+			return "", fmt.Errorf("found multiple compute agentPoolProfiles")
+		}
+
+		app = &cs.Properties.AgentPoolProfiles[i]
```
Comment (asalkeld) [inline pkg/config/derive.go]: you could break here

**Comment by asalkeld** [comment]
/retest

a connection timeout failure

File: pkg/config/derive.go (hunk 1 above)
Comment (jim-minter) [reply pkg/config/derive.go]: no, because the if test above would never succeed in that case

**Comment by mjudeikis** [comment]
lgtm

**Comment by jim-minter** [comment]
marking lgtm based on above

**Pull request merged**